### PR TITLE
Add Rental repository and service test implementation

### PR DESCRIPTION
**Pull request by nklimovych**

### CODE REVIEW

File: src/test/java/online/carsharing/service/impl/RentalServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+        manager.setId(MANAGER_ID);
+        manager.setRoles(Set.of(managerRole));
+
+        customer = new User();
+        customer.setId(CUSTOMER_ID);
+        customer.setRoles(Set.of(customerRole));
+
+        car = new Car();
+        car.setId(CAR_ID);
+        car.setInventory(2);
```
Comment (SkillfulDev) [inline src/test/java/online/carsharing/service/impl/RentalServiceImplTest.java]:
```suggestion
        car.setInventory(INVENTORY_AMOUNT);
```
200w

File: src/test/java/online/carsharing/service/impl/RentalServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    @DisplayName("getUserRentals retrieves active and inactive rentals of all users by manager")
+    public void getUserRentals_ManagerRetrievesAllActiveAndInactive_ListOfRentalResponseDto() {
+        List<Rental> activeRentals = generateRentals(true, ACTIVE_RENTALS_AMOUNT);
+        List<Rental> inactiveRentals = generateRentals(false, INACTIVE_RENTALS_AMOUNT);
+
+        when(rentalRepository.findAllByActive(true)).thenReturn(activeRentals);
+        when(rentalRepository.findAllByActive(false)).thenReturn(inactiveRentals);
+
+        List<RentalResponseDto> allActiveRentals =
+                rentalService.getUserRentals(manager, null, true);
```
Comment (SkillfulDev) [inline src/test/java/online/carsharing/service/impl/RentalServiceImplTest.java:208]: I don’t know if it’s a good practice to put null

**Review by SkillfulDev — DISMISSED** [review]

File: src/test/java/online/carsharing/service/impl/RentalServiceImplTest.java (hunk 2 above)
Comment (nklimovych) [reply src/test/java/online/carsharing/service/impl/RentalServiceImplTest.java:208]: The `userId` is optional in the `getUserRentals()` method, which is why it is part of the logic

**Review by HAIOVYI — DISMISSED** [review]
GJ!

**Review by SkillfulDev — APPROVED** [review]

**Review by azzilian — APPROVED** [review]

**Pull request merged**